### PR TITLE
provide import.meta.env.SITE when there are private envs

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- When we have discovered private envs (which is all of the time, baically), still provide `import.meta.env.SITE`.
- Closes https://github.com/withastro/astro/issues/3489

## Testing

Test added

## Docs

N/A, bug fix

### CODE REVIEW

File: packages/astro/src/vite-plugin-env/index.ts [hunk 1]
```diff
@@ -85,6 +85,7 @@ export default function envVitePlugin({
 					replacements = Object.fromEntries(entries);
 					// These additional replacements are needed to match Vite
 					replacements = Object.assign(replacements, {
+						'import.meta.env.SITE': astroConfig.site ? `'${astroConfig.site}'` : 'undefined',
```
[the remark this answers is not in the document]
Comment (matthewp) [reply packages/astro/src/vite-plugin-env/index.ts:88]: I can add a test about destructing and make sure we handle that. What do you mean by whitespace?

[the remark this answers is not in the document]
Comment (natemoo-re) [reply packages/astro/src/vite-plugin-env/index.ts:88]: Actually never mind, I think just making sure we handle destructuring is fine.

**Comment by matthewp** [comment]
@natemoo-re updated to account for destructing, and test added for that.

**Review by bholmesdev — APPROVED** [review]
Ah, I forgot about destructuring support when I added this feature... `env`aluable catch 👏

**Review by natemoo-re — APPROVED** [review]
LGTM!

**Pull request merged**